exercises/concept/paolas-prestigious-pizza/.meta/Exemplar.elm

```diff
@@ -0,0 +1,92 @@
+module PaolasPrestigiousPizza exposing
+    ( Pizza
+    , ingredientsParser
+    , menuParser
+    , oneIngredientParser
+    , pizzaParser
+    , priceParser
+    , vegetarianParser
+    , wordParser
+    )
+
+import Parser exposing ((|.), (|=), Parser)
+
+
+type alias Pizza =
+    { name : String
+    , vegetarian : Bool
+    , ingredients : List String
+    , price : Int
+    }
+
+
+priceParser : Parser Int
+priceParser =
+    Parser.int
+        |. Parser.symbol "€"
+
+
+vegetarianParser : Parser Bool
+vegetarianParser =
+    Parser.oneOf
+        [ Parser.map (always True) (Parser.symbol "(v)")
+        , Parser.succeed False
+        ]
+
+
+wordParser : Parser String
+wordParser =
+    Parser.chompWhile Char.isAlpha
+        |> Parser.getChompedString
+
+
+ingredientsParser : Parser (List String)
+ingredientsParser =
+    sequenceParser "," oneIngredientParser
+
+
+pizzaParser : Parser Pizza
+pizzaParser =
+    Parser.succeed Pizza
+        |= wordParser
+        |. Parser.spaces
+        |= vegetarianParser
+        |. Parser.spaces
+        |. Parser.symbol ":"
+        |. Parser.spaces
+        |= ingredientsParser
+        |. Parser.spaces
+        |. Parser.symbol "-"
+        |. Parser.spaces
+        |= priceParser
+
+
+menuParser : Parser (List Pizza)
+menuParser =
+    sequenceParser "" pizzaParser |. Parser.end
+
+
+oneIngredientParser : Parser String
```

Comment:
I would move this function up, so it is just below ingredientsParser

Reply:
Yes, it's more natural.
In the stub I will leave it where it is because the order follows the tasks.

Reply:
The tasks grow in complexity, this one is the most complex because it uses `andThen`.
At first, the students use `wordParser` for ingredients too and parse pizzas that way, then I introduce multi-word ingredients and they have to write this specialized parser.

I did it this way because I wanted the students to be able to parse a simple pizza as quick as possible and get some satisfaction/motivation along the way. After that, you refine the code to handle edge cases.